packages/serverpod/lib/src/relic/web_server.dart

```diff
@@ -164,7 +164,12 @@ class WebServer {
       logError(e, stackTrace: stackTrace);
 
       request.response.statusCode = HttpStatus.internalServerError;
-      request.response.write('$e');
+      if (serverpod.runMode == ServerpodRunMode.production) {
```

Comment:
For now we just keep the error internally on production deployments, but I also wondered whether this should be configurable.

Then we would need to somehow set it to enabled for development (where it's very helpful), but should make extra sure that consumers never enable it in production.

Relying on the "Run mode" enum here seems best, as we would be aware when changing this in the future (as there have been discussions to have more flexible run modes).

Reply:
Suggestion: Remove the production mode check, I would like to move away from doing hard checks on the mode we are running in. This should mostly be used by the application layer and selection of configuration.

We are already logging the exception so the programmer would have access to the error during development.

Reply:
@Isakdl So we would then move this into the configuration? E.g. something like `returnErrorsOnWebserver: true`?

> could you also check if we are still throwing the null check cast error if you delete the template folder as described in the issue.

The `!` has been removed a while back, and now we return a specific error message (like "template not found") in that case.

But still I think the overall issue was not that specific error message or the `!` (which is not really worse than doing the null check ourselves and then throwing an exception), but rather the leaking of any internals, which is now prevented.

Reply:
@Isakdl Okay, to summarize your suggestion would be to never return the “raw error message” (even during development), as it’s visible in the logs and printed to the console.

Makes sense to me as well, and then we don’t add another dependency on the obsolete mode. And indeed, the logs would be the first thing the developer looks at anyway when encountering such an error.

Will change this then accordingly.